### ISSUE TITLE
[SPIR-V] unexpected builtin `RayTmaxKHR` for `RayTCurrent` when `HitTKHR` was expected

### ISSUE THREAD

**Issue by farzonl** [microsoft/DirectXShaderCompiler#6739]
The Docs indicate that calls to `RayTCurrent` should trigger the Builtin `HitTKHR`, but the builtin  `RayTmaxKHR` is used in its place in the example below:
```hlsl

[shader("intersection")]
void fn() {
float ret = RayTCurrent();
}
```

Either the docs below or the code needs to be updated I'm not sure which.

https://github.com/microsoft/DirectXShaderCompiler/blob/main/docs/SPIR-V.rst

| HLSL                               | SPIR-V | HLSL Shader Stage                                                                              | 
|---------------------------------|-----------|-------------------------------------------------------------------------------------------|

| System Value Intrinsic | Builtin | Raygen | Intersection | Closest Hit | Any Hit | Miss | Callable |
|----------------------------------|---------------------|------------|-------------------|-----------------|-------------|-------------|-------------|
| RayTCurrent()                | HitT{NV/KHR} | ✓           | ✓                   | ✓                 | ✓           |                |                 |

Invocation command:
```bash
DXC_Debug_BUILD/bin/dxc scratch/WorldRayDirection_spirv_test.hlsl -T lib_6_8 -spirv -fspv-target-env=universal1.5 -O0 -fcgl
```

Result:
```spirv
; SPIR-V
; Version: 1.5
; Generator: Google spiregg; 0
; Bound: 14
; Schema: 0
               OpCapability MinLod
               OpCapability FragmentShaderSampleInterlockEXT
               OpCapability FragmentShaderPixelInterlockEXT
               OpCapability FragmentShaderShadingRateInterlockEXT
               OpCapability ComputeDerivativeGroupQuadsNV
               OpCapability ComputeDerivativeGroupLinearNV
               OpCapability RayQueryKHR
               OpCapability VulkanMemoryModelDeviceScope
               OpCapability GroupNonUniformPartitionedNV
               OpCapability RayTracingKHR
               OpCapability VulkanMemoryModel
               OpExtension "SPV_EXT_fragment_shader_interlock"
               OpExtension "SPV_NV_compute_shader_derivatives"
               OpExtension "SPV_KHR_ray_query"
               OpExtension "SPV_KHR_vulkan_memory_model"
               OpExtension "SPV_NV_shader_subgroup_partitioned"
               OpExtension "SPV_KHR_ray_tracing"
               OpMemoryModel Logical Vulkan
               OpEntryPoint IntersectionKHR %fn "fn" %2
               OpSource HLSL 680
               OpName %fn "fn"
               OpName %src_fn "src.fn"
               OpName %bb_entry "bb.entry"
               OpName %ret "ret"
               OpDecorate %2 BuiltIn RayTmaxKHR
      %float = OpTypeFloat 32
%_ptr_Input_float = OpTypePointer Input %float
       %void = OpTypeVoid
          %6 = OpTypeFunction %void
%_ptr_Function_float = OpTypePointer Function %float
          %2 = OpVariable %_ptr_Input_float Input
         %fn = OpFunction %void None %6
          %7 = OpLabel
          %8 = OpFunctionCall %void %src_fn
               OpReturn
               OpFunctionEnd
     %src_fn = OpFunction %void None %6
   %bb_entry = OpLabel
        %ret = OpVariable %_ptr_Function_float Function
         %13 = OpLoad %float %2
               OpStore %ret %13
               OpReturn
               OpFunctionEnd
```

**Comment by s-perron** [comment]
We should update the doc. HitTKHR was renamed to RayTMaxKHR. See [here](https://htmlpreview.github.io/?https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/KHR/SPV_KHR_ray_tracing.html#:~:text=Removed%20HitTKHR%20alias%20of%20RayTmaxKHR).
